core/src/test/java/hudson/util/io/ZipArchiverTest.java

```diff
@@ -2,173 +2,84 @@
-            /* We probably don't have enough free disk space
-             * That's ok, we'll skip this test...
-             */
-            LOGGER.log(Level.SEVERE, "Couldn't set up huge file for huge64bitFile test", e);
-            return;
+            // We probably don't have enough free disk space. That's ok, we'll skip this test...
+            Assume.assumeNoException(e);
         }
 
         // a file to store the zip archive in
-        File zipFile = null;
+        Path zipFile = Files.createTempFile(tmp.getRoot().toPath(), "test", ".zip");
 
         // create zip from tmpDir
-        ZipArchiver archiver = null;
-
-        try {
-            zipFile = File.createTempFile("test", ".zip");
-            archiver = new ZipArchiver(Files.newOutputStream(zipFile.toPath()));
-
-            archiver.visit(hugeFile, "huge64bitFileTest.txt");
-        } catch (Exception e) {
-            fail("exception driving ZipArchiver", e);
-        } finally {
-            if (archiver != null) {
-                try {
-                    archiver.close();
-                } catch (IOException e) {
-                    // ignored
-                }
-            }
+        try (ZipArchiver archiver = new ZipArchiver(Files.newOutputStream(zipFile))) {
+            archiver.visit(hugeFile.toFile(), "huge64bitFileTest.txt");
         }
 
         // examine zip contents and assert that there's an item there...
-        String zipEntryName = null;
-
-        try (ZipFile zipFileVerify = new ZipFile(zipFile)) {
-
-            zipEntryName = ((ZipEntry) zipFileVerify.entries().nextElement()).getName();
-        } catch (Exception e) {
-            fail("failure enumerating zip entries", e);
-        }
-
-        assertEquals("huge64bitFileTest.txt", zipEntryName);
-    }
-
-    /**
-     * Convenience method for failing with a cause.
-     *
-     * @param msg the failure description
-     * @param cause the root cause of the failure
-     */
-    private void fail(final String msg, final Throwable cause) {
-        LOGGER.log(Level.SEVERE, msg, cause);
-        Assert.fail(msg);
-    }
-
-    /**
-     * Recursively deletes a directory and all of its children.
-     *
-     * @param f the File (neé, directory) to delete
-     */
-    private void deleteDir(final File f) {
-        for (File c : f.listFiles()) {
-            if (c.isDirectory()) {
-                deleteDir(c);
+        try (ZipFile zipFileVerify = new ZipFile(zipFile.toFile())) {
+            assertEquals(1, zipFileVerify.size());
+            ZipEntry zipEntry = zipFileVerify.entries().nextElement();
+            assertEquals("huge64bitFileTest.txt", zipEntry.getName());
+            assertEquals(length, zipEntry.getSize());
+        } catch (ZipException e) {
+            if (e.getMessage().contains("invalid CEN header (bad signature)")) {
+                // Probably running on OpenJDK 8 and hitting JDK-8186464
+                Assume.assumeNoException(e);
```

Comment:
I got this error on OpenJDK 1.8.0_282 but not OpenJDK 11.0.10 (Ubuntu). I think it is a likely case of [JDK-8186464](https://bugs.openjdk.java.net/browse/JDK-8186464), which was not backported to OpenJDK 8.

Reply:
I think its hard to claim that we support ZIP64 if this doesn't work on JDK8 since I would guess a significant portion of jenkins users are on JDK8. We might have to reword the changelog.

Perhaps a workaround can be found or we can convince the maintainers of the jdk to backport their fix to rectify this issue for all.

Reply:
But as of this PR `hudson.util.io.ZipArchiver` (the only production code being modified in this PR) _does_, when running on OpenJDK 8, generate valid ZIP64 archives that can be extracted with any valid ZIP64 decompression tool (including Info-ZIP).

Reply:
Ah I misunderstood, thanks for clearing it up